src/runtime/node/internal/perf_hooks/performance.ts

```diff
@@ -37,11 +33,123 @@ const nodeTiming = {
   },
 } satisfies nodePerfHooks.PerformanceNodeTiming;
 
-// Performance
-export const Performance = class Performance
-  extends _Performance<nodePerfHooks.PerformanceEntry>
-  implements nodePerfHooks.Performance
+// PerformanceEntry
+export class PerformanceEntry implements nodePerfHooks.PerformanceEntry {
+  readonly __unenv__ = true;
+  detail: any | undefined;
+  entryType = "event" as any; // abstract
+  name: string;
+  startTime: number;
+  constructor(name: string, options?: PerformanceMarkOptions) {
+    this.name = name;
+    this.startTime = options?.startTime || performance.now();
+    this.detail = options?.detail;
+  }
+  get duration(): number {
+    return performance.now() - this.startTime;
+  }
+  toJSON() {
+    return {
+      name: this.name,
+      entryType: this.entryType,
+      startTime: this.startTime,
+      duration: this.duration,
+      detail: this.detail,
+    };
+  }
+}
+
+// PerformanceMark
+export const PerformanceMark: typeof nodePerfHooks.PerformanceMark = class PerformanceMark
+  extends PerformanceEntry
+  implements nodePerfHooks.PerformanceMark
 {
+  entryType = "mark" as const;
+
+  constructor() {
```

Comment:
Is this constructor needed? 
If so should it takes ...args?

Reply:
not needed but typescript is nagging (it was before)